src/main/java/friday/logic/Parser.java

```diff
@@ -0,0 +1,123 @@
+package friday.logic;
+
+import friday.model.Deadline;
+import friday.model.ToDo;
+import friday.model.Event;
+import friday.ui.Ui;
+import friday.storage.Storage;
+import friday.exception.FridayException;
+import friday.model.TaskList;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Contains the logic of {@link friday.app.Friday}
+ * Checks for proper formatting of input, if improper
+ * a {@link friday.exception.FridayException} will be thrown
+ * Else the stored list of tasks will be modified
+ * according to the input, either by adding or removing
+ * a task, or marking a task as complete/incomplete.
+ */
+public class Parser {
+    private static final DateTimeFormatter OUT_DATE = DateTimeFormatter.ofPattern("MMM d yyyy");
+    private static final DateTimeFormatter OUT_DT   = DateTimeFormatter.ofPattern("MMM d yyyy, h:mma");
+
+    /** yyyy-MM-dd[ HHmm] or d/M/yyyy[ HHmm]; date-only -> 00:00 */
+    public static LocalDateTime parseDT(String s) throws FridayException {
+        s = s.trim();
+        for (String p : new String[]{"yyyy-MM-dd HHmm","yyyy-MM-dd","d/M/yyyy HHmm","d/M/yyyy"}) {
+            DateTimeFormatter f = DateTimeFormatter.ofPattern(p);
+            try {
+                return p.contains("HHmm") ? LocalDateTime.parse(s, f) : LocalDate.parse(s, f).atStartOfDay();
+            } catch (DateTimeParseException ignored) {}
+        }
+        throw new FridayException("Cannot parse date/time. Use yyyy-MM-dd[ HHmm] or d/M/yyyy[ HHmm].");
+    }
+    static String fmt(LocalDateTime dt) {
```

Comment:
Should the method name be renamed to more clearly be a verb? Also, should it be public or private?
```suggestion
    static String format(LocalDateTime dt) {
```